ingress/nginx.go

```diff
@@ -33,43 +40,63 @@ type loadBalancerTemplate struct {
 	Entries []LoadBalancerEntry
 }
 
-func (lb *nginxLoadBalancer) configLocation() string {
-	return lb.conf.ConfigDir + "/nginx.conf"
+func (lb *nginxLoadBalancer) nginxConfFile() string {
+	return lb.WorkingDir + "/nginx.conf"
 }
 
 // NewNginxLB creates a new LoadBalancer
 func NewNginxLB(nginxConf NginxConf) LoadBalancer {
+	nginxConf.WorkingDir = strings.TrimSuffix(nginxConf.WorkingDir, "/")
 	return &nginxLoadBalancer{
-		conf:      nginxConf,
+		NginxConf: nginxConf,
 		signaller: &DefaultSignaller{}}
 }
 
 func (lb *nginxLoadBalancer) Start() error {
-	// Write out initial configuration
-	lb.update(LoadBalancerUpdate{Entries: []LoadBalancerEntry{}})
+	err := lb.initialiseNginxConf()
+	if err != nil {
+		return fmt.Errorf("unable to initialise nginx config: %v", err)
+	}
 
-	cmd := exec.Command(lb.conf.BinaryLocation, "-c", lb.configLocation())
+	lb.cmd = exec.Command(lb.BinaryLocation, "-c", lb.nginxConfFile())
 
-	cmd.Stdout = log.StandardLogger().Writer()
-	cmd.Stderr = log.StandardLogger().Writer()
-	cmd.Stdin = os.Stdin
+	lb.cmd.Stdout = log.StandardLogger().Writer()
+	lb.cmd.Stderr = log.StandardLogger().Writer()
+	lb.cmd.Stdin = os.Stdin
 
-	lb.cmd = cmd
-	err := cmd.Start()
+	log.Info("(Ignore errors about /var/log/nginx/error.log - they are expected)")
+	err = lb.cmd.Start()
 	if err != nil {
 		return fmt.Errorf("unable to start nginx: %v", err)
 	}
 
+	lb.running.set(true)
+
 	go func() {
-		log.Info("Nginx has exited: ", cmd.Wait())
+		log.Info("Nginx has exited: ", lb.cmd.Wait())
+		lb.running.set(false)
 	}()
 
-	log.Debugf("Nginx pid %d", cmd.Process.Pid)
+	time.Sleep(nginxStartDelay)
+	if !lb.running.get() {
+		return fmt.Errorf("nginx died shortly after starting")
+	}
+
+	log.Debugf("Nginx pid %d", lb.cmd.Process.Pid)
 	return nil
 }
 
+func (lb *nginxLoadBalancer) initialiseNginxConf() error {
+	err := os.Remove(lb.nginxConfFile())
+	if err != nil {
+		log.Debugf("Can't remove nginx.conf: %v", err)
+	}
+	_, err = lb.update(LoadBalancerUpdate{Entries: []LoadBalancerEntry{}})
+	return err
+}
+
 func (lb *nginxLoadBalancer) Stop() error {
-	log.Infof("Shutting down process %d", lb.cmd.Process.Pid)
```

Comment:
Was this causing a race? This was useful to know when I ended up with a zombie :)